src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1275,9 +1313,97 @@ bool entitiesQuery
         cerV->push_back(cer);
     }
 
+    /* All the not-patterned entities in the request not in the response are added (without attributes), as they are
+     * used before prunning in the CPr calculation logic */
+    for (unsigned int ix = 0; ix < enV.size(); ++ix)
+    {
+      if (enV.get(ix)->isPattern != "true")
+      {
+        bool needToAdd = true;
+        for (unsigned int jx = 0; jx < cerV->size(); ++jx)
+        {
+          if (cerV->get(jx)->contextElement.entityId.id == enV.get(ix)->id && cerV->get(jx)->contextElement.entityId.type == enV.get(ix)->type)
```

Comment:
I would put 'cerV->get(jx)->contextElement.entityId.id == enV.get(ix)->id' within parenthesis, and also of course, what's to the right of '&&', for better readibility